Cython/Compiler/FlowControl.py

```diff
@@ -1016,7 +1016,11 @@ def mark_forloop_target(self, node):
             # Python strings, etc., while correctly falling back to an
             # object type when the base type cannot be handled.
 
-            self.mark_assignment(target, node.item)
+            self.mark_assignment(target, ExprNodes.IndexNode(
+                node.pos,
+                base=sequence,
+                index=ExprNodes.IntNode(target.pos, value='PY_SSIZE_T_MAX',
+                                        type=PyrexTypes.c_py_ssize_t_type)))
```

Comment:
This is the implementation used in TypeInference.py. I am not quite sure why it was changed in https://github.com/cython/cython/commit/0b547d5cf87eed72039ee727deea6d9539ec1898. However I assume it should be the same in both implementations. This fixes #3787

Reply:
We're still in alpha phase and 0b547d5 didn't add a test, so … ;-)